guix-science-nonfree/packages/nvidia-container-toolkit.scm

```diff
@@ -0,0 +1,203 @@
+;;; WITHOUT ANY WARRANTY; without even the implied warranty of
+;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;;; GNU General Public License for more details.
+;;;
+;;; You should have received a copy of the GNU General Public License
+;;; along with GNU Guix.  If not, see <http://www.gnu.org/licenses/>.
+
+(define-module (guix-science-nonfree packages nvidia-docker)
+  #:use-module (guix gexp)
+  #:use-module (guix git-download)
+  #:use-module (guix utils)
+  #:use-module (guix build utils)
+  #:use-module (guix packages)
+  #:use-module (guix build-system go)
+  #:use-module (guix build-system gnu)
+  #:use-module (guix licenses)
+  #:use-module (gnu packages)
+  #:use-module (gnu packages docker)
+  #:use-module (gnu packages commencement)
+  #:use-module (gnu packages onc-rpc)
+  #:use-module (gnu packages golang)
+  #:use-module (gnu packages check)
+  #:use-module (gnu packages base)
+  #:use-module (gnu packages curl)
+  #:use-module (gnu packages version-control)
+  #:use-module (gnu packages pkg-config)
+  #:use-module (gnu packages gcc)
+  #:use-module (gnu packages tls)
+  #:use-module (gnu packages elf)
+  #:use-module (gnu packages m4)
+  #:use-module (gnu packages linux)
+  #:use-module (nongnu packages nvidia)
+  )
+
+(define-public nvidia-modprobe
+  (package
+    (name "nvidia-modprobe")
+    (version "550.54.14")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                    (url "https://github.com/NVIDIA/nvidia-modprobe")
+                    (commit version)))
+              (file-name (git-file-name name version))
+              (sha256
+               (base32 "1a7q03pnwk3wa0p57whwv2mvz60bv77vvvaljqzwnscpyf94q548"))))
+    (build-system gnu-build-system)
+    (arguments
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+	  (delete 'configure)
+          (add-before 'build 'set-correct-cflags
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+    	      (setenv "CFLAGS" "-fPIC")
+	      (display "setting CFLAGS\n")
+       	      (substitute* "modprobe-utils/nvidia-modprobe-utils.c"
+		(("^static int nvidia_cap_get_device_file_attrs")
+		 "int nvidia_cap_get_device_file_attrs"))))
+	  (add-after 'build 'build-static-link-libraries
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+	      (invoke "ar" "rcs" "_out/Linux_x86_64/libnvidia-modprobe-utils.a" "_out/Linux_x86_64/nvidia-modprobe-utils.o" "_out/Linux_x86_64/pci-sysfs.o")
+	      (copy-recursively "_out/Linux_x86_64/" (string-append (assoc-ref %outputs "out") "/lib"))))
+	  (delete 'check)
+	  (add-after 'patch-source-shebangs 'replace-prefix
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+    	      (setenv "CC" "gcc")
+	      (setenv "PREFIX" (assoc-ref %outputs "out"))
+	      (copy-recursively "modprobe-utils/" (string-append (assoc-ref %outputs "out") "/include"))
+	      #true) ; must return true for success
+	    ))
```

Comment:
It is no longer necessary to return`#true` from succeeding phases, so I'd recommend removing it from here and elsewhere.

I'd also suggest running `guix style -f nvidia-container-toolkit.scm` to adjust formatting.